### DISCUSSION TITLE
How can I get the current changeid for scripting purpose?

### DISCUSSION THREAD

**Discussion by jbcdnr** [jj-vcs/jj#4049] [Q&A] [answered]
I would like to store the current change id in a variable to script something like:
```bash
jj git fetch
CURRENT_CHANGEID=...
jj new main
jj move --from $CURRENT_CHANGEID a_file
jj describe -m 'Description'
jj git push -c @
```

The closest I could get is `jj log -l 1 --template 'self.change_id()'` but there is still the leading `◉  ` in front. Is there a better way to do this?

Thank you!

**Comment by jbcdnr** [comment]
A workaround
```bash
jj new
jj move --from @- a_file
jj describe -m 'Description'
jj git fetch
jj rebase -r @ -d main
jj git push -c @
```
and there is probably a way to do it with `split`.

**Reply by arxanas** [reply]
Also feel free to discuss at https://github.com/martinvonz/jj/issues/3814 if you have thoughts.

**Comment by yuja** [comment]
Use `log --no-graph` to suppress graph output. And yes, `split` and `rebase` will probably work.

**Reply by jbcdnr** [reply, marked as answer]
Thank you for pointing me to`--no-graph`. Complete solution:
```bash
jj log -l 1 --template 'self.change_id()' --no-graph
```

**Reply by martinvonz** [reply]
`-r @` is better than `-l 1` because I don't think the first commit is necessarily the current commit.

**Reply by ilyagr** [reply]
I'd also add `--color=never`.
